src/helm/benchmark/presentation/summarize.py

```diff
@@ -403,20 +404,23 @@ def get_stats_file_metadata(data_overlap_dir: str) -> Dict[str, List[str]]:
                 data_overlap_stats_key = data_overlap_stats.data_overlap_stats_key
                 light_scenario_key = data_overlap_stats_key.light_scenario_key
                 scenario_spec = light_scenario_key.scenario_spec
-                num_instances = data_overlap_stats.num_instances
                 n = data_overlap_stats_key.overlap_protocol_spec.n
-                """
-                TODO: here we are currently just aggregating across all instance ids
-                for a given scenario rather than the subset run on HELM
-                """
-                num_overlapping_inputs = len(data_overlap_stats.instance_ids_with_overlapping_input)
-                num_overlapping_references = len(data_overlap_stats.instance_ids_with_overlapping_reference)
-                if n == OVERLAP_N_COUNT:
-                    scenario_spec_overlap_counts[scenario_spec] = (
-                        num_instances,
-                        num_overlapping_inputs,
-                        num_overlapping_references,
+                if scenario_spec in self.scenario_spec_instance_id_dict:
+                    # Get statistics based on the subset of instance_ids that HELM uses for a scenario
+                    instance_ids = self.scenario_spec_instance_id_dict[scenario_spec]
+                    num_instances = len(instance_ids)
+                    num_overlapping_inputs = len(
+                        set(data_overlap_stats.instance_ids_with_overlapping_input) & set(instance_ids)
                     )
+                    num_overlapping_references = len(
+                        set(data_overlap_stats.instance_ids_with_overlapping_reference) & set(instance_ids)
+                    )
+                    if n == OVERLAP_N_COUNT:
```

Comment:
Can you do `if n != OVERLAP_N_COUNT: continue` right underneath `for data_overlap_stats in data_overlap_stats_list:`? i.e. exit early.